src/Grpc.Net.Client/Balancer/Internal/ChildHandlerLoadBalancer.cs

```diff
@@ -0,0 +1,248 @@
+                        // Load balancing config is not what we're currently using.
+
+                        // Already have a pending load balancer. Dispose it and start over.
+                        _pending?.LoadBalancer.Dispose();
+                        _pending = CreateLoadBalancer(factory, state.LoadBalancingConfig);
+                    }
+                }
+                else
+                {
+                    throw new InvalidOperationException($"Couldn't resolve load balancing policy {state.LoadBalancingConfig.PolicyName} to a factory");
+                }
+            }
+
+            lock (_lock)
+            {
+                if (_pending == null)
+                {
+                    if (_current == null)
+                    {
+                        if (_initialServiceConfig != null && _initialServiceConfig.LoadBalancingConfigs.Count > 0)
+                        {
+                            if (TryGetValidServiceConfigFactory(_initialServiceConfig.LoadBalancingConfigs, _connectionManager.LoadBalancerFactories, out var lbConfig, out var factory))
+                            {
+                                _current = CreateLoadBalancer(factory, lbConfig);
+                            }
+                            else
+                            {
+                                throw new InvalidOperationException($"No load balancers configured for {string.Join(", ", _initialServiceConfig.LoadBalancingConfigs.Select(c => $"'{c.PolicyName}'"))}.");
+                            }
+                        }
+                        else if (TryGetFactory(LoadBalancingConfig.PickFirstPolicyName, _connectionManager.LoadBalancerFactories, out var defaultFactory))
+                        {
+                            // Default to pick_first when no configuration supplied.
+                            _current = CreateLoadBalancer(defaultFactory, DefaultConfig);
+                        }
+                        else
+                        {
+                            throw new InvalidOperationException("Unable to create default load balancer.");
+                        }
+                    }
+
+                    childToUpdate = _current.Value;
+                }
+                else
+                {
+                    childToUpdate = _pending.Value;
+                }
+            }
+
+            childToUpdate.LoadBalancer.UpdateChannelState(state);
+        }
+
+        private (LoadBalancer, string) CreateLoadBalancer(LoadBalancerFactory factory, LoadBalancingConfig lbConfig)
+        {
+            var controller = new ChildHandlerController(this);
+            var loadBalancer = factory.Create(controller, (IDictionary<string, object>)lbConfig.Inner[lbConfig.PolicyName]);
+            var result = (loadBalancer, lbConfig.PolicyName);
+            controller.Child = result;
+
+            return result;
+        }
+
+        public static bool TryGetValidServiceConfigFactory(
```

Comment:
So this just resolves the first match? Is it possible for there to be multiple matches?

Reply:
Yes. We're looking for the first match. That is standard behavior. Added link to doc.